include/device.h

```diff
@@ -404,10 +409,15 @@ static inline int device_set_power_state(const struct device *dev,
 static inline int device_get_power_state(const struct device *dev,
 					 uint32_t *device_power_state)
 {
-	return dev->device_pm_control(dev,
-					 DEVICE_PM_GET_POWER_STATE,
-					 device_power_state,
-					 NULL, NULL);
+	if (dev->device_pm_control) {
+		return dev->device_pm_control(dev,
+						 DEVICE_PM_GET_POWER_STATE,
+						 device_power_state, NULL, NULL);
+	} else {
+		return device_pm_control_nop(dev,
```

Comment:
I think it would be better to get rid of `device_pm_control_nop()` function, replace `device_pm_control_nop` use with `NULL` and simply return `-ENOTSUP` here.

Reply:
Possibly, but this all has to be revisited anyway.  I remember at one point there were problems with the power management functions returned an error, which is part of why the nop implementation exists.